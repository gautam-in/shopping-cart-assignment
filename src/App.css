body{
    font-family: 'Dosis', sans-serif;;
    /* padding: 20px 60px; */
}

a {
    text-decoration: none;
    color: black;
    cursor: pointer;
}

* {
    box-sizing: border-box;
}

