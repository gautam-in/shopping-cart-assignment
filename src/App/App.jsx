import React from 'react';
import { Router, Route, Switch, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import { history } from '../_helpers';
import { alertActions } from '../_actions';
import { PrivateRoute } from '../_components';
import { HomePage } from '../HomePage';
import { LoginPage } from '../LoginPage';
import { RegisterPage } from '../RegisterPage';

class App extends React.Component {
    constructor(props) {
        super(props);

        history.listen((location, action) => {
            // clear alert on location change
            this.props.clearAlerts();
        });
    }

    render() {
        const { alert } = this.props;
        return (
            <div className="jumbotron">
                <div className="container">
                    <div className="col-sm-8 col-sm-offset-2">
                        
                        <Router history={history}>
                            <Switch>
                                <PrivateRoute exact path="/" component={HomePage} />
                                <Route path="/login" component={LoginPage} />
                                <Route path="/register" component={RegisterPage} />
                                <Redirect from="*" to="/" />
                            </Switch>
                        </Router>
                    </div>
                </div>
                {alert.message &&
                            <p style={{color:'red',paddingLeft:'62%'}}>{alert.message}</p>
                }
            </div>
        );
    }
}

function mapState(state) {
    const { alert } = state;
    return { alert };
}

const actionCreators = {
    clearAlerts: alertActions.clear
};

const connectedApp = connect(mapState, actionCreators)(App);
export { connectedApp as App };