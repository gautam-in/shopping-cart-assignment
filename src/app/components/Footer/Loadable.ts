/**
 *
 * Asynchronously loads the component for Footer
 *
 */

import { lazyLoad } from 'utils/loadable';

export const Footer = lazyLoad(
  () => import('./index'),
  module => module.Footer,
);
