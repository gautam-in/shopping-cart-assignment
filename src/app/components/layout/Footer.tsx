import React from 'react';

const Footer = () => {
    return (
        <div className='bg-gray-200 fixed bottom-0 h-8 w-full text-center p-1 text-gray-600'  >
            Copyright @2011-2023 Sabka Bazaar Grocery Supplies Pvt. Ltd.
        </div>
    );
}

export default Footer;