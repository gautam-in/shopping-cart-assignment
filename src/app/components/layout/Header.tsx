import React from "react";
import Link from "next/link";
import Image from "next/image";
import { useSelector, useDispatch } from "react-redux";

const Header = () => {
  let cartState = useSelector((state: any) => state.cart);
  console.log("cartState  from store : ", cartState);
  let cartText = "-";
  if (cartState) {
    cartText = `${cartState.totalItems} ${
      cartState.totalItems > 1 ? "items" : "item"
    }`;
  }

  return (
    <nav className=" h-28 shadow-lg">
      <div className="max-w-screen-lg mx-auto h-full flex justify-between items-center">
        <Link href="/offers" className="">
          <Image
            src="/static/images/logo_2x.png"
            width="150"
            height="150"
            alt="Logo Image"
          />
        </Link>
        <div className="hidden h-full md:flex lg:flex">
          <ul className="flex flex-col md:flex-row lg:flex-row h-full gap-4 items-end pb-4 text-gray-500 cursor-pointer  font-extrabold">
            <li>
              <Link href="/offers"> Home </Link>
            </li>
            <li>
              <Link href="/products">Products</Link>
            </li>
          </ul>
        </div>

        <div className="h-full flex flex-col pt-5 ">
          <ul className=" flex gap-4 mb-2 cursor-pointer">
            <li>
              <Link href="/login">SignIn</Link>
            </li>
            <li>
              <Link href="/logout">SignOut</Link>
            </li>
            <li>
              <Link href="/signup">Register</Link>
            </li>
            <li>
              <button
                data-collapse-toggle="mobile-menu"
                type="button"
                className="lg:hidden md:hidden inline-flex items-center p-2 ml-1 text-sm text-gray-500 rounded-lg hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200"
              >
                <span className="sr-only">Open main menu</span>
                <svg
                  className="w-6 h-6"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    fill-rule="evenodd"
                    d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                    clip-rule="evenodd"
                  ></path>
                </svg>
              </button>
            </li>
          </ul>
          <div className=" bg-[#ddd] flex-1 text-center py-2 cursor-pointer">
            <Image
              src="/static/images/cart1.svg"
              alt="cart"
              height={40}
              width={40}
              className="inline-block"
            />
            <span className="inline-block pl-2">{cartText}</span>
          </div>
        </div>
      </div>
    </nav>
  );
};

export default Header;
