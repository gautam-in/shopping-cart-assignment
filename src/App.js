import { Routes, Route } from 'react-router-dom';
import './App.css';
import Home from './pages/Home';
import 'bootstrap/dist/css/bootstrap.min.css';
import TopNavbar from './navigation/TopNavbar';
import Plpage from './pages/Plpage';
import RegisterForm from './components/register/RegisterForm'
import LoginComp from './components/login/LoginComp';
import FooterComp from './components/footer/FooterComp';

function App() {
  return (
    <>
    <Routes>
      {/* <Route path="/" element={<TopNavbar/>}> */}
        <Route index element={<Home />} />
        <Route path='plp' element={<Plpage />} />
        <Route path='signup' element={<RegisterForm />} />
        <Route path='login' element={<LoginComp />} />
      {/* </Route> */}

      
    </Routes>
    <FooterComp />
    </>
    
    
  );
}

export default App;
