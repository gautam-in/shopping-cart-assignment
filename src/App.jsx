import React, { lazy, Suspense } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import "./App.scss";
import ErrorBoundary from "./Component/ErrorBoundary/errorBoundary.component";
import Footer from "./Component/Footer/footer.component";
import Header from "./Component/Header/header.component";
import Spinner from "./Component/Spinner/spinner.component";
import CartContainer from "./Screen/Cart/cart.container";

const ProductPage = lazy(() => import("./Screen/Product/product.container"));
const LoginPage = lazy(() => import("./Screen/Login/login.container"));
const HomePage = lazy(() => import("./Screen/Home/home.container"));
const RegisterPage = lazy(() => import("./Screen/Register/register.conatiner"));

function App() {
  return (
    <Router>
      <div className="app-container">
        <Header />
        <Switch>
          <ErrorBoundary>
            <Suspense fallback={<Spinner />}>
              <Route path="/" exact component={HomePage} />
              <Route path="/product" component={ProductPage} />
              <Route path="/login" exact component={LoginPage} />
              <Route path="/register" exact component={RegisterPage} />
            </Suspense>
          </ErrorBoundary>
        </Switch>
        <CartContainer />
        <Footer />
      </div>
    </Router>
  );
}

export default App;
