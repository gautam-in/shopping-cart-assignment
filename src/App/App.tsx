import { Route, Routes } from "react-router";
import { lazy, Suspense, useEffect, memo } from "react";
import { loadBanners, loadCategories, loadProduct } from "../Model/Model";
import { FooterComponent } from "../BaseComponent/Footer/FooterComponent";
import { HeaderComponent } from "../BaseComponent/Header/HeaderComponent";
const HomeComponent = lazy(() => import("../Pages/Home/HomeComponent"));
const SignInComponent = lazy(() => import("../Pages/SignIn/SignInComponent"));
const RegisterComponent = lazy(
    () => import("../Pages/Register/RegisterComponent")
);

const ProductComponent = lazy(
    () => import("../Pages/Product/ProductComponent")
);
const CartPageComponent = lazy(() => import("../Pages/Cart/CartPageComponent"));
const Page404Component = lazy(
    () => import("../Pages/Page404/Page404Component")
);

export const App: React.FC = memo(
    () => {
        useEffect(() => {
            loadBanners();
            loadCategories();
            loadProduct();
        }, []);
        
        return (
            <>
                <HeaderComponent />
                <Suspense fallback={<div>Page Is Loading...</div>}>
                    <Routes>
                        <Route path="/" element={<HomeComponent />} />
                        <Route path="signin" element={<SignInComponent />} />
                        <Route path="register" element={<RegisterComponent />} />
                        <Route path="product" element={<ProductComponent />} />
                        <Route path="cart" element={<CartPageComponent />} />
                        <Route path="*" element={<Page404Component />} />
                    </Routes>
                </Suspense>
                <FooterComponent />
            </>
        );
    }
)
