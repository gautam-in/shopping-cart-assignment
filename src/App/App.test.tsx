import { ShallowWrapper, shallow } from "enzyme";
import { FooterComponent } from "../BaseComponent/Footer/FooterComponent";
import { HeaderComponent } from "../BaseComponent/Header/HeaderComponent";
import { App } from "./App";

describe("App", () => {
    const tree: ShallowWrapper = shallow(<App/>);
    test("App Snapshot render", () => {
        expect(tree).toMatchSnapshot();
    });

    test("Check Header and Footer Loaded Snapshot render", () => {
        expect(tree.containsAllMatchingElements([
            <HeaderComponent />,
            <FooterComponent />
            ])).toEqual(true);
    });
});
