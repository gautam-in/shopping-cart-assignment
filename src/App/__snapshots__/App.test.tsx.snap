// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Snapshot render 1`] = `ShallowWrapper {}`;
